packages/gatsby/src/internal-plugins/partytown/gatsby-browser.tsx

```diff
@@ -1,64 +1,24 @@
-import React, { ReactElement, useState } from "react"
-import type { GatsbySSR } from "gatsby"
-import { Partytown } from "@builder.io/partytown/react"
-import { PartytownContext, ScriptProps } from "gatsby-script"
+import { collectedScriptsByPage } from "gatsby-script"
+import { injectPartytownSnippet } from "./utils/inject-partytown-snippet"
+import type { GatsbyBrowser } from "gatsby"
 
-interface ICollectedForwardsState {
-  collectedForwards: Set<string>
-  collectedAnyScript: boolean
-}
-
-function PartytownProvider({ children }): ReactElement {
-  const [{ collectedForwards, collectedAnyScript }, setState] =
-    useState<ICollectedForwardsState>({
-      collectedForwards: new Set(),
-      collectedAnyScript: false,
-    })
+export const onInitialClientRender: GatsbyBrowser[`onInitialClientRender`] =
+  (): void => {
+    injectPartytownSnippet(window.location.pathname)
```

Comment:
Why not pass the scripts directly here?
```suggestion
    injectPartytownSnippet(collectedScriptsByPage.get(window.location.pathname))
```